lib/features/mailbox/data/repository/mailbox_repository_impl.dart

```
@@ -48,51 +48,39 @@ class MailboxRepositoryImpl extends MailboxRepository {
-            mailboxCacheList: localMailboxResponse.mailboxes!);
-
-        await Future.wait([
-          mapDataSource[DataSourceType.local]!.update(
-              accountId,
-              session.username,
-              updated: newMailboxUpdated,
-              created: changesResponse.created,
-              destroyed: changesResponse.destroyed),
-          if (changesResponse.newStateMailbox != null)
-            stateDataSource.saveState(accountId, session.username, changesResponse.newStateMailbox!.toStateCache(StateType.mailbox)),
-        ]);
-      }
-    } else {
-      final mailboxResponse = await mapDataSource[DataSourceType.network]!.getAllMailbox(session, accountId);
-
-      await Future.wait([
-        mapDataSource[DataSourceType.local]!.update(accountId, session.username, created: mailboxResponse.mailboxes),
-        if (mailboxResponse.state != null)
-          stateDataSource.saveState(accountId, session.username, mailboxResponse.state!.toStateCache(StateType.mailbox)),
-      ]);
+    if (localMailboxResponse.hasMailboxes) {
+      log('MailboxRepositoryImpl::getAllMailbox: MAILBOX_CACHED = ${localMailboxResponse.mailboxes?.length} | STATE_CACHED = ${localMailboxResponse.state?.value}');
+      yield localMailboxResponse;
     }
 
+    final networkMailboxResponse = await mapDataSource[DataSourceType.network]!
+      .getAllMailbox(session, accountId);
+
+    log('MailboxRepositoryImpl::getAllMailbox: MAILBOX_NETWORK = ${networkMailboxResponse.mailboxes?.length} | STATE_NETWORK = ${networkMailboxResponse.state?.value}');
+
+    await Future.wait([
+      mapDataSource[DataSourceType.local]!.update(
+        accountId,
+        session.username,
+        created: networkMailboxResponse.mailboxes
+      ),
+      if (networkMailboxResponse.state != null)
+        stateDataSource.saveState(
+          accountId,
+          session.username,
+          networkMailboxResponse.state!.toStateCache(StateType.mailbox)
+        ),
+    ]);
```

we can ignore error for this functions